### PR TITLE
 added default status and default message functionality

### PR DESCRIPTION
**Pull request by heinrich10**
* added eslint
* added unit test
* added code coverage

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+/coverage/**
+/docs/**
+/jsdoc/**
+/templates/**
+/tests/bench/**
+/tests/fixtures/**
+/tests/performance/**
+/tmp/**
+/lib/util/unicode/is-combining-character.js
+test.js
```
Comment (harunurhan) [inline .eslintignore]: I wouldn't ignore the tests

File: index.js [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
-}
-module.exports = AmkError;
+
+const { checkStatusCode } = require('./lib/check');
+
+const defaultMessage = 'Something went wrong';
+
+class AmkError extends Error {
+
+	constructor(message, status) {
```
Comment (harunurhan) [inline index.js]: `message = defaultMessage` instead of `||`

File: lib/check.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
+const checkStatusCode = (status) => {
```
Comment (harunurhan) [inline lib/check.js]: I don't think this name is good for what it's doing. It seems to me that it is more like `sanitizeStatusCode`

File: package.json [hunk 4]
```diff
@@ -1,10 +1,11 @@
 {
   "name": "amk-error",
-  "version": "0.1.0",
+  "version": "0.2.0",
   "description": "http response error object for amk js",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "lint": "./node_modules/.bin/eslint .",
```
Comment (harunurhan) [inline package.json]: should be `./node_module/.bin/` by default, so you can just `eslint .`

File: package.json [hunk 5]
```diff
@@ -1,10 +1,11 @@
...
+    "test": "./node_modules/.bin/nyc ./node_modules/mocha/bin/_mocha"
```
Comment (harunurhan) [inline package.json]: ditto

File: lib/check.js (hunk 3 above)
Comment (heinrich10) [reply lib/check.js]: makes sense, thanks for this

**Pull request merged**